home/import_ordered_content_sets.py

```diff
@@ -0,0 +1,227 @@
+        page_slugs: list[str],
+        contact_fields: list[str],
+    ) -> None:
+        """
+        Given the extracted values from a row of the file, create the corresponding ordered content set pages.
+
+        Note that this adds the pages to the OrderedContentSet object, hence there is no return value.
+
+        :param ordered_set: The ordered content set to add the pages to.
+        :param times: The times extracted from the row.
+        :param units: The units extracted from the row.
+        :param before_or_afters: The before or afters extracted from the row.
+        :param page_slugs: The page slugs extracted from the row.
+        :param contact_fields: The contact fields extracted from the row.
+        """
+        for idx, page_slug in enumerate(page_slugs):
+            if not page_slug or page_slug == "-":
+                continue
+            page = ContentPage.objects.filter(slug=page_slug).first()
+            time = times[idx]
+            unit = units[idx]
+            before_or_after = before_or_afters[idx]
+            contact_field = contact_fields[idx]
+            if page:
+                ordered_set.pages.append(
+                    (
+                        "pages",
+                        {
+                            "contentpage": page,
+                            "time": time or "",
+                            "unit": unit or "",
+                            "before_or_after": before_or_after or "",
+                            "contact_field": contact_field or "",
+                        },
+                    )
+                )
+            else:
+                logger.warning(f"Content page not found for slug '{page_slug}'")
+
+    def _create_ordered_set_from_row(
+        self, index: int, row: dict[str, str]
+    ) -> OrderedContentSet:
+        """
+        Create an ordered content set from a single row of the file.
+
+        :param index: The row number of the row being processed.
+        :param row: The row of the file as a dictionary.
+        :return: An instance of OrderedContentSet.
+        :raises ImportException: If time, units, before_or_afters, page_slugs and contact_fields are not all equal length.
+        """
+        ordered_set = self._get_or_init_ordered_content_set(row, row["Name"])
+
+        times = self._csv_to_list(row["Time"])
+        units = self._csv_to_list(row["Unit"])
+        before_or_afters = self._csv_to_list(row["Before Or After"])
+        page_slugs = self._csv_to_list(row["Page Slugs"])
+        # backwards compatiblilty if there's only one contact field
+        contact_fields = row["Contact Field"].split(",")
+        contact_fields = (
+            [p.strip() for p in contact_fields]
+            if len(contact_fields) > 1
+            else [contact_fields[0]] * len(times)
+        )
+
+        self._validate_extracted_values(
+            index, row, times, units, before_or_afters, page_slugs, contact_fields
```

Comment:
Instead of passing these around separately a bunch of times, could we make a class to hold them and move all the logic related to parsing and such onto that?